### PR TITLE
Updated livescript for PID fingers tuning and related scripts

### PR DESCRIPTION
**Pull request by simeonedussoni**
in this PR some improvements are implemented, the most important:
- interactive choice of the data files for calibration
- save of the report in dedicated file with speaking name together with `.m` dataset
- script to reset workspace

### CODE REVIEW

File: src/tools/ergocub-fingers-tuning/init_ws.m
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/init_ws.m:1]: Missing Copyright

File: src/tools/ergocub-fingers-tuning/results/middle6-cus_PID - Copy.pdf
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/results/middle6-cus_PID - Copy.pdf:1]: Strage name for the file. Better off repurposing or simply delete it if it's a leftover.

File: src/tools/ergocub-fingers-tuning/results/middle7-cus_PID - Copy.pdf
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/results/middle7-cus_PID - Copy.pdf:1]: Same as above.

File: src/tools/ergocub-fingers-tuning/results/rinky2-cus_PID - Copy.pdf
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/results/rinky2-cus_PID - Copy.pdf:1]: Same as above.

File: src/tools/ergocub-fingers-tuning/results/rinky3-cus_PID - Copy.pdf
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/results/rinky3-cus_PID - Copy.pdf:1]: Same as above.

File: src/tools/ergocub-fingers-tuning/scripts/design_robust_pid.m [hunk 1]
```diff
@@ -4,16 +4,41 @@
...
+            C = tunablePID('C', 'PID');
+            C.Kp.Minimum = -inf;    C.Kp.Maximum = 0;
+            C.Ki.Minimum = -inf;    C.Ki.Maximum = 0;
+            C.Kd.Minimum = -inf;    C.Kd.Maximum = 0;
+            C.Tf.Minimum = 0.01;   C.Tf.Maximum = 0.1;    % N = 1/Tf
+            %disp('PID')
+            assignin('base','tYpe','PID');
+
+    otherwise 
+        disp('scegli PI o PID')
```
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/scripts/design_robust_pid.m:36]: No Italian, please.

File: src/tools/ergocub-fingers-tuning/scripts/Reshape_dataset.m [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+% All Rights Reserved
+% Authors: mattia.fussi@iit.it, simeone.dussoni@iit.it
+%
+function t = Reshape_dataset(file, tau, downsample, interpola)
+
+    arguments
+        file {mustBeFile}
+        tau (1, 1) {mustBeNonnegative}
+        downsample (1,1) {mustBeInteger}
+        interpola (1,1) {mustBeInteger}
```
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/scripts/Reshape_dataset.m:11]: No Italian, please.

File: src/tools/ergocub-fingers-tuning/README.md [hunk 3]
```diff
@@ -47,4 +47,11 @@ Copy the `.csv` datasets created with `ident-finger` in the `data` folder. Then:
 1. Add to the path the folders `data` and `scripts`
 2. Open the live script `identification_tuning.mlx`
 3. Check the file names of the training and validation sets
+   - the file opened is the last one, unless the user run **init_ws** which resets the file name
+   - **init_ws** must also be run as the first command to insure the livescript works.
+   - it's better to also 
+   - it's better to also 'clear all output' by right-clicking in the livescript otherwise the report can be filled with old data and plots!
```
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/README.md:53]: It seems there's a typo.

File: src/tools/ergocub-fingers-tuning/README.md [hunk 4]
```diff
@@ -47,4 +47,11 @@ Copy the `.csv` datasets created with `ident-finger` in the `data` folder. Then:
...
 4. Run it by pressing "Run" or F5
+   - sometimes (very often) the final command 'export' doesn't produce a complete pdf file. re-run this line by selecting it until the pdf looks fine
```
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/README.md:55]: Uhm 🤔
This should be fixed upfront. Fine for now, but then the root cause ought to be sorted out.

**Review by pattacini — CHANGES_REQUESTED** [review]
Some inline comments to address.

File: src/tools/ergocub-fingers-tuning/README.md [hunk 5]
```diff
@@ -47,4 +47,10 @@ Copy the `.csv` datasets created with `ident-finger` in the `data` folder. Then:
 1. Add to the path the folders `data` and `scripts`
 2. Open the live script `identification_tuning.mlx`
 3. Check the file names of the training and validation sets
+   - the file opened is the last one, unless the user run **init_ws** which resets the file name
+   - **init_ws** must also be run as the first command to insure the livescript works.
+   - it's better to also 'clear all output' by right-clicking in the livescript otherwise the report can be filled with old data and plots!
 4. Run it by pressing "Run" or F5
+   - sometimes (very often) the final command 'export' doesn't produce a complete pdf file. re-run this line by selecting it until the pdf looks fine 
+5. check the plots: anytime the fingers are not moving, the PID parameters are screwed up see e.g. [here](https://github.com/icub-tech-iit/ergocub-design-lowerarm/issues/245#issuecomment-2095469706)
```
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/README.md:55]:
```suggestion
5. check the plots: if the fingers are not moving, then the PID gains are most likely not up to the task; e.g., see [here](https://github.com/icub-tech-iit/ergocub-design-lowerarm/issues/245#issuecomment-2095469706).
```

File: src/tools/ergocub-fingers-tuning/scripts/init_ws.m [hunk 6]
```diff
@@ -0,0 +1,11 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Copyright (C) 2024 Fondazione Istitito Italiano di Tecnologia (IIT)
+% All Rights Reserved.
```
Comment (pattacini) [inline src/tools/ergocub-fingers-tuning/scripts/init_ws.m:3]:
```suggestion
% All Rights Reserved.
% Authors: simeone.dussoni@iit.it
%
```

**Review by pattacini — CHANGES_REQUESTED** [review]
Some more hints.

I'll leave the review of the live script to @mfussi66.

**Review by pattacini — APPROVED** [review]
I've just merged my suggested edits straight away.
Fine with me 👍🏻 

Awaiting @mfussi66's review.

**Review by mfussi66 — CHANGES_REQUESTED** [review]
Thanks for the PR! 
I suggest the following to avoid putting too many files:
- remove all the .csv files that are not actually used for the tuning
- remove the pdf files, looking at the live script and launching it should be sufficient
- very minor, maybe replace the .mat files with simple text files?

**Comment by simeonedussoni** [comment]
I left only the most useful  couple of matlab files to save the values of Kp,Ki and Kd for the fingers.
Also, I comented the lines to produce the .m file and the .pdf report (an eye check can be done when running the livescript that the plots are ok and the step response also)

**Comment by mfussi66** [comment]
Apologies for the delay, thanks for the changes @simeonedussoni ! 

I opened the script and it works well, the only thing I would change is adding the `init_ws` call at the beginning of the live script.

**Pull request closed without merging**